### DISCUSSION TITLE
How can I create a classLibrary for TemplateControls?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10633] [Q&A] [answered]
I want to customize some controls and make them reusable for any other project. I will add some dependencyproperty and rewrite its UI.

I don't know how to achieve this but I did this like WPF custom control library.

In Visual Studio Avalonia has not any TemplateControls Library Template so I have to create a .net standard 2.1 class library.

Soon I add Avalonia nuget package and create a new TemplateControls.

Then I created a new Avalonia project and referenced the ClassLibrary.

Now the problem comes, in WPF I can find its namespace in xmlns and use it directly, but now I can find nothing namespace about it.

What's wrong with it?

**Comment by rabbitism** [comment]
You should be able to find the xmlns namespace.

**Comment by kyuranger** [comment, marked as answer]
I found the solution by the source code of Aura.UI(https://github.com/PieroCastillo/Aura.UI/tree/master/src/Aura.UI.FluentTheme)

Firstly, create a templatedcontrol and set its axaml file to AvaloniaResource.

Secondly, create an Avalonia Style and include all the templatedcontrols inside it. just like this:
<Styles xmlns="https://github.com/avaloniaui"		
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
		x:Class="Sample.GlobalStyle">
  <Design.PreviewWith>
    <Border Padding="20">
      
    </Border>
  </Design.PreviewWith>

	<StyleInclude Source="/Controls/Button.axaml" />
</Styles>

Tertiary, create a class for x:class in the style just created:
using Avalonia.Styling;
using System;
using System.Collections.Generic;
using System.Text;

namespace Sample
{
    public class GlobalStyle:Style
    {
    }
}

Then, reference your class library in the project you want and add these codes below in app.axaml.
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"			 
             x:Class="Sample.App"
             RequestedThemeVariant="Default">
             

    <Application.Styles>
        <FluentTheme />
		<StyleInclude Source="avares://Sample/GlobalStyle.axaml"/>		
	</Application.Styles>
</Application>

Now you can use the templatedcontrol in any where you want.
